src/System.Linq/src/System/Linq/Range.cs

```diff
@@ -101,6 +101,18 @@ public List<int> ToList()
                 return list;
             }
 
+            public HashSet<int> ToHashSet(IEqualityComparer<int> comparer)
+            {
+                HashSet<int> hashSet = new HashSet<int>(_end - _start, comparer);
```

Comment:
I think this is one of the few places where you can keep the count, since `Range` returns distinct numbers. Although, in the case of a weird comparer that says everything is equal we would still over-allocate. /cc @JonHanna